File: Algorithms/C++/Tree/226_Invert_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == NULL) {
            return NULL;
        }

        TreeNode* inverted_left = invertTree(root->left);
        TreeNode* inverted_right = invertTree(root->right);

        root->left = inverted_right;
        root->right = inverted_left;

        return root;
    }
};

File: Algorithms/Swift/Tree/226_Invert_Binary_Tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */
class Solution {
    func invertTree(root: TreeNode?) -> TreeNode? {
        if root == nil {
            return nil
        }

        let left = invertTree(root!.left)
        let right = invertTree(root!.right)

        root!.left = right
        root!.right = left

        return root
    }
}